### PR TITLE
fix: api injection (VF-000)

### PR DESCRIPTION
**Pull request by DecathectZero**
**Fixes or implements VF-000**

### Brief description. What is this change?

I saw this error after bumping the general-runtime package for alexa:
```
Error: env var: GENERAL_SERVICE_ENDPOINT not found
```
the alexa-runtime definitely doesn't need the `GENERAL_SERVICE_ENDPOINT` to work. Because we referenced `@/config` directly, the downstream runtimes now need the same environment variables.

Stems from: https://github.com/voiceflow/general-runtime/pull/255

Instead, they should be instantiated as the handlers get being added.
This is blocking some critical bug fixes for alexa.

`API_HANDLER_ENDPOINT` is no longer needed, this was a reference to the API microservice. It is done locally now.

### Related PRs

| branch              | PR          |
| ------------------- | ----------- |
| other_pr_production | [link](url) |
| other_pr_master     | [link](url) |

### Checklist

- [ ] this is a breaking change and should publish a new major version
- [ ] appropriate tests have been written

### CODE REVIEW

File: config.ts [hunk 1]
```diff
@@ -31,10 +31,9 @@ const CONFIG: Config = {
   // integrations block
   INTEGRATIONS_HANDLER_ENDPOINT: getOptionalProcessEnv('INTEGRATIONS_HANDLER_ENDPOINT') || 'none',
   // api-block
-  API_HANDLER_ENDPOINT: getOptionalProcessEnv('API_HANDLER_ENDPOINT'),
-  API_MAX_TIMEOUT_MS: Number(getOptionalProcessEnv('API_MAX_TIMEOUT_MS', '20000')),
-  API_MAX_CONTENT_LENGTH_BYTES: Number(getOptionalProcessEnv('API_MAX_CONTENT_LENGTH_BYTES', '1000000')),
-  API_MAX_BODY_LENGTH_BYTES: Number(getOptionalProcessEnv('API_MAX_BODY_LENGTH_BYTES', '1000000')),
+  API_MAX_TIMEOUT_MS: Number(getOptionalProcessEnv('API_MAX_TIMEOUT_MS')) || null,
```
Comment (DecathectZero) [inline config.ts:34]: `Number(null | undefined)` results in `NaN` we still want `null`

File: runtime/lib/Handlers/api/index.ts [hunk 2]
```diff
@@ -26,7 +22,7 @@ const APIHandler: HandlerFactory<Node.Integration.Node, IntegrationsOptions | vo
         actionBodyData.headers = [...headers, { key: USER_AGENT_KEY, val: USER_AGENT }];
       }
 
-      const data = await makeAPICall(actionBodyData, runtime);
+      const data = await makeAPICall(actionBodyData, runtime, config as ResponseConfig);
```
Comment (DecathectZero) [inline runtime/lib/Handlers/api/index.ts:25]: so it won't resolve `undefined` with `void` so I have to cast it.

Comment (jonahsnider) [inline runtime/lib/Handlers/api/index.ts:25]: this type assertion scares me, should `makeAPICall` be updated to allow `Partial<ResponseConfig>`? i think that would fix the types here

File: tests/runtime/lib/Handlers/api.unit.ts [hunk 3]
```diff
@@ -40,8 +38,8 @@ describe('API Handler unit tests', () => {
       sinon.restore();
     });
     it('success', async () => {
-      const customAPIEndpoint = 'https://foo';
-      const apiHandler = APIHandler({ customAPIEndpoint });
+      const config = 'config';
```
Comment (jonahsnider) [inline tests/runtime/lib/Handlers/api.unit.ts:41]: surely this should be `{}` and not a string

**Review by jonahsnider — APPROVED** [review]

File: tests/runtime/lib/Handlers/api.unit.ts (hunk 3 above)
Comment (DecathectZero) [reply tests/runtime/lib/Handlers/api.unit.ts:41]: it doesn't matter too much, I'm just checking that whatever I set it to is getting passed directly into the function and not getting modified.

File: runtime/lib/Handlers/api/utils.ts [hunk 4]
```diff
@@ -9,7 +9,6 @@ import querystring from 'querystring';
 import safeJSONStringify from 'safe-json-stringify';
 import validator from 'validator';
 
-import CONFIG from '@/config';
```
Comment (DecathectZero) [inline runtime/lib/Handlers/api/utils.ts:12]: this was breaking downstream runtimes

File: config.ts (hunk 1 above)
Comment (jonahsnider) [reply config.ts:34]: yeah JS is funny in that sense.
null gets casted to 0, undefined gets casted to NaN

we should really come up with some util for converting optional env vars to numbers

Comment (trs) [reply config.ts:34]: @DecathectZero @jonahsnider This package is super helpful for reading and converting environment variables: https://www.npmjs.com/package/env-var

**Pull request merged**